### PR TITLE
semgrep 0.22.0

### PR DESCRIPTION
**Pull request by mschwager**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by mschwager** [comment]
It's not clear to me why CI is failing. It looks like something to do with `ruamel`. Do any maintainers have an idea?

**Comment by SMillerDev** [comment]
The python part of the build failed. You should check if all resources are still pointing to the latest available version.

**Comment by brendongo** [comment]
@SMillerDev updated python resources but it still fails with the same error locally. This wasn't an error when running locally early yesterday. Any ideas?

**Comment by brendongo** [comment]
The specific error goes away when I remove the resource calls for ruamel.yaml and ruamel.yaml.clib but those two packages have not been updated since Feb2020. (brew test fails since we are missing a needed dependency) and these two have no other dependencies so not sure what else would have changed?

**Comment by brendongo** [comment]
Looks like setuptools released a new major version and it was not playing well with ruamel: https://github.com/pypa/setuptools/issues/2355 looking if we can do any of the suggested fixes on formula level.

**Comment by SMillerDev** [comment]
Is there a pull request or an issue for the specific module opened yet?

**Comment by brendongo** [comment]
I found this: https://sourceforge.net/p/ruamel-yaml/tickets/356/ should I mention it in the comments?

**Comment by brendongo** [comment]
FWIW the fix will be on the setuptools side and they provided an escape hatch as a way for tools to keep working while they iron out their bugs so we can remove the env variable when  pypa/setuptools#2355  is closed

File: Formula/semgrep.rb [hunk 1]
```diff
@@ -128,6 +112,11 @@ def install
       # Used by semgrep-core for clang to find libtree-sitter.a
       ENV["LIBRARY_PATH"] = lib
 
+      # Officially suggested workaround for breaking change in setuptools v50.0.0
+      # See: https://setuptools.readthedocs.io/en/latest/history.html#v50-0-0
```
Comment (SMillerDev) [inline Formula/semgrep.rb:116]: Can you put the Sourceforge ticket here instead. I care more about the solution than the problem.

Comment (brendongo) [reply Formula/semgrep.rb]: Done!

**Review by SMillerDev — APPROVED** [review]
Thanks @mschwager  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**